### PR TITLE
Cherrypick various configuration and docs fixes to 0.3.0 release branch

### PR DESCRIPTION
**Pull request by davidz627**
Fixes: #183

/assign @msau42

### CODE REVIEW

File: test/k8s-integration/main.go [hunk 1]
```diff
@@ -205,8 +205,8 @@ func runTests(k8sDir string) error {
 	if err != nil {
 		return err
 	}
-	testArgs := "--test_args=--ginkgo.focus=CSI\\sdriver:.*gcePD-external"
-	cmd := exec.Command("go", "run", "hack/e2e.go", "--", "--check-version-skew=false", "--test", testArgs)
+	testArgs := "--test_args=--ginkgo.focus=CSI.*gcePD-external --ginkgo.skip=BlockVolume|\\[Serial\\]|kubelet.*down"
```
Comment (msau42) [inline test/k8s-integration/main.go]: We don't need to skip BlockVolume anymore now that it's beta.

Comment (davidz627) [reply test/k8s-integration/main.go]: should be fixed in master branch then cherry-picked back to v0.3.0. I'm not going to modify these commits for now

Comment (davidz627) [reply test/k8s-integration/main.go]: oh looks like it has been changed

**Comment by davidz627** [comment]
@msau42 comments addressed

**Comment by davidz627** [comment]
@msau42 missed the cherrypick of the prow head overlays so the integration tests were failing

File: deploy/kubernetes/overlays/prow-head-template/node_images.yaml [hunk 2]
```diff
@@ -9,4 +9,8 @@ spec:
         - name: gce-pd-driver
           image: PROW_GCEPD_IMAGE:latest
         - name: csi-driver-registrar
-          image: gcr.io/gke-release/csi-driver-registrar:v0.4.1-gke.0
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-head-template/node_images.yaml]: ?

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-head-template/node_images.yaml]: very weird, my editor wasn't showing that as a thing. but when i went in with vim I could see the merge conflict. Fixed

File: Makefile [hunk 3]
```diff
@@ -18,8 +18,8 @@
 STAGINGIMAGE=${GCE_PD_CSI_STAGING_IMAGE}
 STAGINGVERSION=${GCE_PD_CSI_STAGING_VERSION}
 
-PRODIMAGE=gcr.io/google-containers/volume-csi/gcp-compute-persistent-disk-csi-driver
-PRODVERSION=v0.2.0.beta
+PRODIMAGE=gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver
```
Comment (msau42) [inline Makefile]: We don't use these settings to create the official images. Should we just remove these lines?

Comment (msau42) [reply Makefile]: I'm worried we may accidentally manually push to it if it's here.

**Pull request merged**